Filter out drivers in Reassign drivers for scheduled rides based on availability.

namanhboi opened the pull request:
### Summary 
When you reassign driver for scheduled rides, it will only show the drivers available. Additionally, you add can more filters into an array called reassignDriverFilters. 

![image](https://github.com/cornell-dti/carriage-web/assets/61220757/b2e5a758-014b-4bc6-a447-3b58d51ce56e)

### Test Plan

I have tested this very informally using console.log but plan to do more rigrous testing with unit tests in the future (after finals).

Atikpui007 commented on frontend/src/components/UserTables/RidesTable.tsx:
```diff
@@ -23,6 +25,62 @@ const RidesTable = ({ rides, hasButtons }: RidesTableProps) => {
+  // You can add additional filters to filter our drivers to reassign here.
+  const reassignDriverFilters: ((driver: DriverType) => boolean)[] = [
+    // (driver: DriverType) => {
+    //   return driver.firstName != 'Bin laden';
```
Please dont add this lol 
Specifically line 42. Change the first name example

namanhboi replied:
Yup, forgot to removed it when I was messing around. Changed it now tho.

raissaji replied:
Don't forget to push your changes 🙏 The test driver name is still there lol

Atikpui007 reviewed (dismissed):
The PR looks good and I like how you used the reassignDriversFilters to allow for more filters when necessary. This was tested and it works as expected 
Good job

namanhboi commented:
I refactored the code to make some functions like isAvaiable (which might be useful in other contexts) exportable and added some needed documentation. Tested everything informatlly and it works. Also resolved any merge conflicts with the master branch at the time of writing this comment.
